### PR TITLE
Bugfix/cam det cpreq to cp

### PR DESCRIPTION
**Pull request by MarcelCaron-NOAA**
## Pull Request Testing ##

- [x] Describe testing already performed for this Pull Request:</br>

This PR removes cpreq usage for copies from /tmp to /com, cam-det component.  No jobs were tested.

---
- [x] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

### (1) Set up jobs 
•	symlink the EVS_fix directory locally as "fix"
•	In all driver scripts, point all exports of "HOMEevs" to your EVS test directory
•	In all stats and plots driver scripts, change COMIN to the parallel directory (`/lfs/h2/emc/vpppg/noscrub/emc.vpppg/$NET/$evs_ver_2d`)

### (2) Test Jobs
Use the following to submit all jobs: `qsub –v vhr=00 `

From your EVS testing directory, the following scripts should be run using the setup in step (1):
```
dev/drivers/scripts/prep/cam/jevs_cam_severe_prep.sh
dev/drivers/scripts/prep/cam/jevs_hireswarw_precip_prep.sh
```
```
dev/drivers/scripts/stats/cam/jevs_cam_hireswarw_snowfall_stats.sh
```
```
dev/drivers/scripts/plots/cam/jevs_cam_headline_plots.sh
```
_[Total: 2 prep jobs; 1 stats job; 1 plots job]_

### (3) Review
- I'll check for red flags:
```
check="FATAL\|WARNING\|error\|Killed\|Cgroup\|argument expected\|No such file\|cannot\|empty\|failed\|unexpected\|exceeded"
grep "$check" $outfile
```

---
- [x] Has the code been checked to ensure that no errors occur during the execution? **NO**

---
- [x] Do these updates/additions include sufficient testing updates? **YES**

---
- [x] Please complete this pull request review by **2/9/2024**.</br>

---
## Pull Request Checklist ##

- [ ] Review the source issue metadata (required labels, projects, and milestone).
- [ ] Complete the PR description above.
- [ ] Ensure the PR title matches the feature branch name.
- [ ] Check the following:
- [ ]  Instructions provided on how to run
- [ ]  Developer's name is replaced by ${user} where necessary throughout the code
- [ ]  Check that the ecf file has all the proper definitions of variables
- [ ]  Check that the jobs file has all the proper settings of COMIN and COMOUT and other input variables
- [ ]  Check to see that the output directory structure is followed
- [ ]  Be sure that you are not using MET utilities outside the METplus wrapper structure

- [ ] After submitting the PR, select **Development** issue with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue.

### CODE REVIEW

File: scripts/plots/cam/exevs_cam_headline_plots.sh [hunk 1]
```diff
@@ -90,7 +90,10 @@ fi
    find ${DATA}/${VERIF_CASE}/* -name "*.png" -type f -print | tar -cvf ${DATA}/${NET}.${STEP}.${COMPONENT}.${RUN}.${VERIF_CASE}.v${VDATE}.tar --transform='s#.*/##' -T -
 
 if [ $SENDCOM = YES ]; then
-    cpreq -v ${DATA}/${NET}.${STEP}.${COMPONENT}.${RUN}.${VERIF_CASE}.v${VDATE}.tar ${COMOUTplots}/.
+    FILE=${DATA}/${NET}.${STEP}.${COMPONENT}.${RUN}.${VERIF_CASE}.v${VDATE}.tar
+    if [ -s "$FILE" ]; then
+       cp -v ${VDATE} ${COMOUTplots}/.
```
Comment (ShelleyMelchior-NOAA) [inline scripts/plots/cam/exevs_cam_headline_plots.sh]: I think ${VDATE} should be ${FILE}.

**Comment by AliciaBentley-NOAA** [comment]
@MarcelCaron-NOAA: Perry will start working on your PR this morning while Shelley works one Perry's two PRs. Keep an eye out for a message from him. Thanks!

@PerryShafran-NOAA @ShelleyMelchior-NOAA

File: scripts/plots/cam/exevs_cam_headline_plots.sh (hunk 1 above)
Comment (PerryShafran-NOAA) [reply scripts/plots/cam/exevs_cam_headline_plots.sh]: @MarcelCaron-NOAA - can we ensure that this line is correct?

**Comment by PerryShafran-NOAA** [comment]
@MarcelCaron-NOAA I have begun working on this PR, starting with the prep jobs.  

You can see the .o files here:  /lfs/h2/emc/vpppg/noscrub/perry.shafran/pr435test/EVS/dev/drivers/scripts/prep/cam

**Comment by PerryShafran-NOAA** [comment]
@MarcelCaron-NOAA The prep jobs are finished.

All output can be seen here:  /lfs/h2/emc/vpppg/noscrub/perry.shafran/evs/v1.0/prep/cam

The precip job looked to be fine, but the severe job has interesting Warnings in there:

```
WARNING: 
WARNING: Zero observations retained!
WARNING: 
```

```
WARNING: 
WARNING: process_point_met_data() -> Invalid GRIB code [Fscale]
WARNING:        But ignored because of empty input
WARNING: 
WARNING: 
WARNING: get_grid_mapping(MET_obs) -> no valid point observation data!
WARNING: 
WARNING: 
WARNING: process_point_met_data() -> No valid data for the variable [Fscale]
WARNING
```

Is there a setting that I didn't set correctly?  Have you seen these warnings before?

**Comment by MarcelCaron-NOAA** [comment]
Hm, those warnings are a little unusual.  Let me check

**Comment by MarcelCaron-NOAA** [comment]
@PerryShafran-NOAA These warnings can occur occasionally.  It means there were no SPC local storm reports on Jan. 29. 2024.  The code will process the empty file and the METplus portion will output a warning.  The job then completes successfully without it.

**Comment by PerryShafran-NOAA** [comment]
@AliciaBentley-NOAA @ShelleyMelchior-NOAA Looks like METplus warnings appear when there are no SPC local storm reports.  We may need to file another bugzilla and report this to DTC to see if they can do something about it.

@MarcelCaron-NOAA In the meantime, these issues are not related to cpreq.  Marcel - please check both prep jobs for the changes from cpreq to cp.  

The stats job is currently running and can be seen here:  /lfs/h2/emc/vpppg/noscrub/perry.shafran/pr435test/EVS/dev/drivers/scripts/stats/cam

**Comment by MarcelCaron-NOAA** [comment]
The jobs completed successfully, their respective outputs look good, and I have no concerns about the log files.

✔️ jevs_cam_hireswarw_precip_prep 
✔️ jevs_cam_severe_prep

**Comment by AliciaBentley-NOAA** [comment]
@PerryShafran-NOAA Thanks for asking about this! WARNING messages _should_ appear when a file is missing! In fact, that might be the only time WARNING messages should appear in EVS v1.0. That is why Wei recently asked people to add WARNING messages for missing files in item #86 in his EVS v1.0 Issues document. If these WARNING messages indicate that SPC local storm reports are missing for a particular day, that is a good thing! Thanks!

@MarcelCaron-NOAA @ShelleyMelchior-NOAA

**Comment by PerryShafran-NOAA** [comment]
@AliciaBentley-NOAA Yes, agreed.  But these Warning messages seem to be a bit more cryptic.  They don't say "Warning, file missing", but they say other things like "Zero observations retained" and "Invalid GRIB code".  If a file is missing, it should be flagged and METplus should be skipped.  

However, in this case, there files exist, but on some days there are no reports.  Here is a listing:

```
-rw-r--r-- 1 dfprod prod   151 Jan 28 06:45 20240120/validation_data/weather/spc/spc_reports_20240120.csv
-rw-r--r-- 1 dfprod prod   151 Jan 29 06:45 20240121/validation_data/weather/spc/spc_reports_20240121.csv
-rw-r--r-- 1 dfprod prod   574 Jan 30 06:45 20240122/validation_data/weather/spc/spc_reports_20240122.csv
-rw-r--r-- 1 dfprod prod   367 Jan 31 06:45 20240123/validation_data/weather/spc/spc_reports_20240123.csv
-rw-r--r-- 1 dfprod prod   269 Feb  1 06:45 20240124/validation_data/weather/spc/spc_reports_20240124.csv
-rw-r--r-- 1 dfprod prod  1991 Feb  2 06:45 20240125/validation_data/weather/spc/spc_reports_20240125.csv
-rw-r--r-- 1 dfprod prod   645 Feb  3 06:45 20240126/validation_data/weather/spc/spc_reports_20240126.csv
-rw-r--r-- 1 dfprod prod  1894 Feb  4 06:45 20240127/validation_data/weather/spc/spc_reports_20240127.csv
-rw-r--r-- 1 dfprod prod   151 Feb  5 06:45 20240128/validation_data/weather/spc/spc_reports_20240128.csv
-rw-r--r-- 1 dfprod prod   151 Feb  6 06:45 20240129/validation_data/weather/spc/spc_reports_20240129.csv
```

When the file is 151 bytes, there are no storm reports, putting up these WARNINGs.  

That said, NCO has probably seen this and it has not been flagged by NCO.  Thus we're probably OK with these warning statements.

**Comment by PerryShafran-NOAA** [comment]
@MarcelCaron-NOAA The stats job is complete.  Please see the stats file here:  /lfs/h2/emc/vpppg/noscrub/perry.shafran/evs/v1.0/stats/cam/atmos.20240204/hireswarw/snowfall.

**Comment by PerryShafran-NOAA** [comment]
The plots job is underway.  See the .o file here:  /lfs/h2/emc/vpppg/noscrub/perry.shafran/pr435test/EVS/dev/drivers/scripts/plots/cam

**Comment by MarcelCaron-NOAA** [comment]
The stats job completed cleanly, the output is comparable to the output in emc.vpppg, and no red flags in the log file. No concerns from me.
✔️ jevs_cam_hireswarw_snowfall_stats

The other job failed at the end.  This is due to the bug Shelley noted earlier.  I pushed a change but I think you're using the code without that change.  Can you update your version of HOMEevs and rerun this job?
❌ jevs_cam_headline_plots

**Comment by AliciaBentley-NOAA** [comment]
@PerryShafran-NOAA We you able to rerun jevs_cam_headline_plots with the bugfix that @MarcelCaron-NOAA made this morning to `cp -v ${FILE} ${COMOUTplots}/.`?

**Comment by PerryShafran-NOAA** [comment]
@AliciaBentley-NOAA @MarcelCaron-NOAA Yes, the plots job is now complete.  Marcel - can you look at the new .o file as well as the output at /lfs/h2/emc/ptmp/perry.shafran/evs/v1.0/plots/cam?

**Comment by MarcelCaron-NOAA** [comment]
The plots job completed cleanly, the output is comparable to recent output in emc.vpppg, and no red flags in the log file. No concerns from me.
✔️ jevs_cam_headline_plots

**Comment by PerryShafran-NOAA** [comment]
Excellent, thanks!  @MarcelCaron-NOAA , do we need a sync fork?  I think not, but please check.

**Comment by PerryShafran-NOAA** [comment]
@AliciaBentley-NOAA do you need to check the code?  It all looks good to me.

**Comment by AliciaBentley-NOAA** [comment]
@PerryShafran-NOAA @MarcelCaron-NOAA I have taken a look at the files that were changed and I think that they look great! The only comment I have is that sometimes we are using "cp -v" and sometimes we are using "cp -vr". I don't know that this makes a dramatic difference, but I wanted to mention it in case it does. If Marcel says that it does not, I approve this PR to be merged!

**Comment by MarcelCaron-NOAA** [comment]
@PerryShafran-NOAA, no, looks like this branch is up to date!

**Comment by PerryShafran-NOAA** [comment]
@MarcelCaron-NOAA @AliciaBentley-NOAA The -r feature in the cp command copies directories recursively (meaning that they copy everything underneath a directory if you are copying a directory.  

I don't think this is needed, because we're copying files, not directories.  But I'd like Marcel to comment about that.  Changing this to just cp -v shouldn't make much of a difference in the cp command if we're copying files.

**Comment by MarcelCaron-NOAA** [comment]
@AliciaBentley-NOAA @PerryShafran-NOAA Yes the -r is for recursive copies and is intentional for the restart copies (cp -rpv)...

...I don't think they are necessary for copying the spatial maps to com, but there will be no difference because we are only copying files.  I haven't tested without the -r option there, but it should work the same.

**Comment by AliciaBentley-NOAA** [comment]
@MarcelCaron-NOAA @PerryShafran-NOAA Thanks! I don't think we need to change anything, especially since we already did the testing, but I wanted to mention it since it was different than other PRs! If you think this PR is ready to merge, then I agree! Thanks!

**Review by PerryShafran-NOAA — APPROVED** [review]
Code works as expected.  Approved for merge.

**Pull request merged**